master/internal/db/postgres_groups.go

```diff
@@ -0,0 +1,218 @@
+	return group, matchSentinelError(err)
+}
+
+// GroupByID looks for a group by id. Returns ErrNotFound if the group isn't found.
+func (db *PgDB) GroupByID(ctx context.Context, gid int) (model.Group, error) {
+	var g model.Group
+	err := Bun().NewSelect().Model(&g).Where("id = ?", gid).Scan(ctx)
+
+	return g, matchSentinelError(err)
+}
+
+// SearchGroups searches the database for groups. userBelongsTo is "optional"
+// in that if a value < 1 is passed in, the parameter is ignored. SearchGroups
+// does not return an error if no groups are found, as that is considered a
+// successful search.
+func (db *PgDB) SearchGroups(ctx context.Context,
+	userBelongsTo model.UserID) ([]model.Group, error) {
+	var groups []model.Group
+	query := Bun().NewSelect().Model(&groups)
+
+	if userBelongsTo > 0 {
+		query = query.
+			Join("INNER JOIN user_group_membership AS ugm ON ugm.group_id=groups.id").
+			Where("ugm.user_id = ?", userBelongsTo)
+	}
+
+	err := query.Scan(ctx)
+
+	return groups, err
+}
+
+// DeleteGroup deletes a group from the database. Returns ErrNotFound if the
+// group doesn't exist.
+func (db *PgDB) DeleteGroup(ctx context.Context, gid int) error {
+	tx, err := Bun().BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		rbErr := tx.Rollback()
+		if rbErr != sql.ErrTxDone && rbErr != nil {
+			log.WithError(rbErr).
+				WithField("groupID", gid).
+				Error("error rolling back transaction in DeleteGroup")
+		}
+	}()
+
+	_, err = tx.NewDelete().
+		Table("user_group_membership").
+		Where("group_id = ?", gid).
+		Exec(ctx)
+	if err != nil {
+		return err
+	}
+
+	res, err := tx.NewDelete().Model(&model.Group{ID: gid}).WherePK().Exec(ctx)
+	if foundErr := mustHaveAffectedRows(res, err); foundErr != nil {
+		return matchSentinelError(foundErr)
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return matchSentinelError(err)
+	}
+
+	return nil
+}
+
+// UpdateGroup updates a group in the database. Returns ErrNotFound if the
+// group isn't found.
+func (db *PgDB) UpdateGroup(ctx context.Context, group model.Group) error {
+	res, err := Bun().NewUpdate().Model(&group).WherePK().Exec(ctx)
+
+	return matchSentinelError(mustHaveAffectedRows(res, err))
+}
+
+// AddUsersToGroup adds users to a group by creating GroupMembership rows.
+// Returns ErrNotFound if the group isn't found or ErrDuplicateRow if one
+// of the users is already in the group.
+func (db *PgDB) AddUsersToGroup(ctx context.Context, gid int, uids ...model.UserID) error {
+	if len(uids) < 1 {
+		return nil
+	}
+
+	groupMem := make([]model.GroupMembership, 0, len(uids))
+	for _, uid := range uids {
+		groupMem = append(groupMem, model.GroupMembership{
+			UserID:  uid,
+			GroupID: gid,
+		})
+	}
+
+	tx, err := Bun().BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		rbErr := tx.Rollback()
+		if rbErr != sql.ErrTxDone && rbErr != nil {
+			log.WithError(rbErr).
+				WithField("groupID", gid).
+				WithField("userIDs", uids).
```

Comment:
nitpick: looking at `git grep 'WithField'` we mostly do `field-names-with-dashes`, e.g. `trial-id` or `allocation-id`
```suggestion
				WithField("group-id", gid).
				WithField("user-ids", uids).
```